src/map/entities/battleentity.cpp

```diff
@@ -733,7 +769,7 @@ void CBattleEntity::SetMLevel(uint8 mlvl)
 
 void CBattleEntity::SetSLevel(uint8 slvl)
 {
-    m_slvl = (slvl > (m_mlvl >> 1) ? (m_mlvl == 1 ? 1 : (m_mlvl >> 1)) : slvl);
+    m_slvl = (slvl > (m_mlvl * map_config.sj_level_ratio) ? (m_mlvl == 1 ? 1 : (m_mlvl * map_config.sj_level_ratio)) : slvl);
```

Comment:
this won't work out in the math the way you think it will. I previously wrote one that will, but its not in the project because of additional work that needed done for the search list to show the correct subjob level at the time.